File: backend/src/main/java/com/smartecommerce/backend/entities/Payment.java
package com.smartecommerce.backend.entities;
import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "payments")
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    @JoinColumn(name = "order_id", nullable = false, unique = true)
    private Order order;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Method method;
    @Column(nullable = false)
    private BigDecimal amount;
    @Enumerated(EnumType.STRING)
    private Status status = Status.PENDING;
    public enum Method { COD, SEPAY_BANK_TRANSFER }
    public enum Status { PENDING, SUCCESS, FAILED }
}

File: frontend/src/services/payment.js
import api from './api';

export const paymentService = {
  getPaymentByOrderId: async (orderId) => {
    const response = await api.get(`/payments/order/${orderId}`);
    return response.data;
  },

  simulateSuccess: async (orderId) => {
    const response = await api.post(`/payments/simulate-success/${orderId}`);
    return response.data;
  }
};
